scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -119,6 +119,67 @@ object TypedPipe extends Serializable {
       override def hash(x: Int): Int = x
     }
   }
+  implicit class HigherPriorityTypedPipeMethods[T, K, V](val tp: TypedPipe[T])(implicit ev: TypedPipe[T] <:< TypedPipe[(K, V)],
+    orderedSerialization: OrderedSerialization[K]) {
+    // scope the imports nice and local
+    import com.twitter.scalding.typed.Sketched
+    import com.twitter.scalding.serialization.{ OrderedSerialization, Serialization }
+    /**
+     * Enables joining when this TypedPipe has some keys with many many values and
+     * but many with very few values. For instance, a graph where some nodes have
+     * millions of neighbors, but most have only a few.
+     *
+     * We build a (count-min) sketch of each key's frequency, and we use that
+     * to shard the heavy keys across many reducers.
+     * This increases communication cost in order to reduce the maximum time needed
+     * to complete the join.
+     *
+     * {@code pipe.sketch(100).join(thatPipe) }
+     * will add an extra map/reduce job over a standard join to create the count-min-sketch.
+     * This will generally only be beneficial if you have really heavy skew, where without
+     * this you have 1 or 2 reducers taking hours longer than the rest.
+     */
+    def sketch(reducers: Int,
+      eps: Double = 1.0E-5, //272k width = 1MB per row
+      delta: Double = 0.01, //5 rows (= 5 hashes)
+      seed: Int = 12345): Sketched[K, V] =
+      Sketched(ev(tp), reducers, delta, eps, seed)(Serialization.toBytes[K](_), orderedSerialization)
+  }
+
+}
+
+private[scalding] trait LowerPriorityTypedPipeImplicits {
+  implicit def toLowerPriorityTypedPipeMethods[T](tp: TypedPipe[T]) =
+    new LowerPriorityTypedPipeMethods(tp)
+}
+private[scalding] class LowerPriorityTypedPipeMethods[T](val tp: TypedPipe[T]) extends AnyVal {
+  // scope the imports nice and local
+  import com.twitter.scalding.typed.Sketched
+  import com.twitter.scalding.serialization.{ OrderedSerialization, Serialization }
+
+  /**
+   * Enables joining when this TypedPipe has some keys with many many values and
+   * but many with very few values. For instance, a graph where some nodes have
+   * millions of neighbors, but most have only a few.
+   *
+   * We build a (count-min) sketch of each key's frequency, and we use that
+   * to shard the heavy keys across many reducers.
+   * This increases communication cost in order to reduce the maximum time needed
+   * to complete the join.
+   *
+   * {@code pipe.sketch(100).join(thatPipe) }
+   * will add an extra map/reduce job over a standard join to create the count-min-sketch.
+   * This will generally only be beneficial if you have really heavy skew, where without
+   * this you have 1 or 2 reducers taking hours longer than the rest.
+   */
+  def sketch[K, V](reducers: Int,
```

Comment:
Once you do, your bound to its method, so the lower priority TypedPipe only being the requirement here means this one is always available. So the errors remain the same if users don't have the implicits for this method for instance